src/main/java/org/prebid/server/bidder/grid/GridBidder.java

```diff
@@ -21,6 +23,22 @@ protected Imp modifyImp(Imp imp, ExtImpGrid impExt) {
         if (impExt.getUid() == null || impExt.getUid() == 0) {
             throw new PreBidException("uid is empty");
         }
+        final GridExtImp gridExtImp = mapper.mapper().convertValue(
+                imp.getExt(),
+                GridExtImp.class
+        );
+
+        String adSlot = getAdSlot(gridExtImp);
```

Comment:
Can be final, but also by saying extracting I mean that we can do something like
```java
final GridExtImp gridExtImp = mapper.mapper().convertValue(imp.getExt(), GridExtImp.class);
final GridExtImpData extImpData = gridExtImp != null ? gridExtImp.getGridExtImpData() : null;
final GridExtImpDataAdServer adServer = extImpData != null ? extImpData.getAdServer() : null;
final String adSlot = adServer != null ? adServer.getAdSlot() : null;

 if (StringUtils.isNotEmpty(adSlot)) {
```